### PR TITLE
Add Python 3.6, Drop Python 3.3

### PR DESCRIPTION
**Pull request by jamadden**
- Badges
- Add more info to README.rst
- zope.testrunner for NS path issues.
- 100% coverage
  - Enable coveralls and a tox environment
  - Use ``with`` to replace ``try/finally`` for simpler code that uses
    fewer lines.
- Whitespace cleanup.
- Modern test assertion methods.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
+=========
+ Changes
+=========
 
 2.2.0 (unreleased)
-------------------
+==================
 
-- Add support for Python 3.5.
+- Add support for Python 3.5 and 3.6.
```
Comment (icemac) [inline CHANGES.rst:8]: `and PyPy3` as this was pending in version 2.1.0.

File: src/zope/ramcache/ram.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 """
 __docformat__ = 'restructuredtext'
```
Comment (icemac) [inline src/zope/ramcache/ram.py:16]: Is this line actually still needed? It seems like cargo-cult to have it at the beginning of many ZF files.

**Review by icemac — APPROVED** [review]
LGTM.

**Comment by icemac** [comment]
@hannosch Could you please add `jamadden` and `icemac` to the PyPI owners of this package?

File: src/zope/ramcache/ram.py (hunk 2 above)
Comment (jamadden) [reply src/zope/ramcache/ram.py:16]: [zope.app.apidoc](https://github.com/zopefoundation/zope.app.apidoc/blob/master/src/zope/app/apidoc/utilities.py#L357) uses it. Up until the most recent release, the default was `structuredtext`, so this was quite necessary.

File: CHANGES.rst (hunk 1 above)
Comment (jamadden) [reply CHANGES.rst:8]: OK.

**Comment by jamadden** [comment]
Thanks.

**Comment by hannosch** [comment]
Added both of you as PyPi owners, happy releasing ;)

**Comment by jamadden** [comment]
Thanks @hannosch !

**Pull request merged**